src/calendar-app/workerUtils/worker/CalendarWorkerLocator.ts

```diff
@@ -0,0 +1,473 @@
+		return new CacheManagementFacade(locator.user, locator.cachingEntityClient, assertNotNull(cache))
+	})
+
+	if (isIOSApp() || isAndroidApp()) {
+		locator.kyberFacade = new NativeKyberFacade(new NativeCryptoFacadeSendDispatcher(worker))
+	} else {
+		locator.kyberFacade = new WASMKyberFacade()
+	}
+
+	locator.pqFacade = new PQFacade(locator.kyberFacade)
+
+	locator.keyLoader = new KeyLoaderFacade(locator.keyCache, locator.user, locator.cachingEntityClient, locator.cacheManagement)
+
+	locator.crypto = new CryptoFacade(
+		locator.user,
+		locator.cachingEntityClient,
+		locator.restClient,
+		locator.rsa,
+		locator.serviceExecutor,
+		locator.instanceMapper,
+		new OwnerEncSessionKeysUpdateQueue(locator.user, locator.serviceExecutor),
+		locator.pqFacade,
+		cache,
+		locator.keyLoader,
+	)
+
+	locator.recoverCode = lazyMemoized(async () => {
+		const { RecoverCodeFacade } = await import("../../../common/api/worker/facades/lazy/RecoverCodeFacade.js")
+		return new RecoverCodeFacade(locator.user, locator.cachingEntityClient, locator.login, locator.keyLoader)
+	})
+	locator.share = lazyMemoized(async () => {
+		const { ShareFacade } = await import("../../../common/api/worker/facades/lazy/ShareFacade.js")
+		return new ShareFacade(locator.user, locator.crypto, locator.serviceExecutor, locator.cachingEntityClient, locator.keyLoader)
+	})
+	locator.counters = lazyMemoized(async () => {
+		const { CounterFacade } = await import("../../../common/api/worker/facades/lazy/CounterFacade.js")
+		return new CounterFacade(locator.serviceExecutor)
+	})
+	locator.groupManagement = lazyMemoized(async () => {
+		const { GroupManagementFacade } = await import("../../../common/api/worker/facades/lazy/GroupManagementFacade.js")
+		return new GroupManagementFacade(
+			locator.user,
+			await locator.counters(),
+			locator.cachingEntityClient,
+			locator.serviceExecutor,
+			locator.pqFacade,
+			locator.keyLoader,
+			await locator.cacheManagement(),
+		)
+	})
+	locator.keyRotation = new KeyRotationFacade(
+		locator.cachingEntityClient,
+		locator.keyLoader,
+		locator.pqFacade,
+		locator.serviceExecutor,
+		cryptoWrapper,
+		locator.recoverCode,
+		locator.user,
+		locator.crypto,
+		locator.share,
+		locator.groupManagement,
+	)
+
+	const loginListener: LoginListener = {
+		onFullLoginSuccess(sessionType: SessionType, cacheInfo: CacheInfo, credentials: Credentials): Promise<void> {
+			return mainInterface.loginListener.onFullLoginSuccess(sessionType, cacheInfo, credentials)
+		},
+
+		onLoginFailure(reason: LoginFailReason): Promise<void> {
+			return mainInterface.loginListener.onLoginFailure(reason)
+		},
+
+		onSecondFactorChallenge(sessionId: IdTuple, challenges: ReadonlyArray<Challenge>, mailAddress: string | null): Promise<void> {
+			return mainInterface.loginListener.onSecondFactorChallenge(sessionId, challenges, mailAddress)
+		},
+	}
+
+	let argon2idFacade: Argon2idFacade
+	if (!isBrowser()) {
```

Comment:
Should we leave the browser check for debugging?

Reply:
Yes, I would like this left in.